packages/autotask-client/src/zip.ts

```diff
@@ -22,7 +22,9 @@ export async function zipFolder(folderPath: string): Promise<string> {
   const zip = new JSZip();
   for (const path of files) {
     const content = await promisify(readFile)(join(folderPath, path));
-    zip.file(path, content);
+    // We hardcode the date so we generate the same zip every time given the same contents
+    // This allows us to use the codedigest to decide whether or not to reupload code
+    zip.file(path, content, { date: new Date(2020, 1, 1, 0, 0, 0, 0) });
```

Comment:
Curious why we need a date here, is codeDigest not only based on contents?

Reply:
Nope! The codeDigest in AWS Lambda is calculated from the zipfile, not from its contents. So the dates of the files within the zipfile actually affect the hash of the zip, and change the codeDigest. I can't explain how many hours I lost to that.

Reply:
> Nope! The codeDigest in AWS Lambda is calculated from the zipfile, not from its contents. So the dates of the files within the zipfile actually affect the hash of the zip, and change the codeDigest. I can't explain how many hours I lost to that.

Jesus. Good to know! Thanks.